src/meshes/p4est_mesh_view.jl

```diff
@@ -0,0 +1,249 @@
+                                                                      parent.unsaved_changes,
+                                                                      parent.current_filename)
+end
+
+@inline Base.ndims(::P4estMeshView{NDIMS}) where {NDIMS} = NDIMS
+@inline Base.real(::P4estMeshView{NDIMS, NDIMS_AMBIENT, RealT}) where {NDIMS, NDIMS_AMBIENT, RealT} = RealT
+
+function extract_p4est_mesh_view(elements_parent,
+                                 interfaces_parent,
+                                 boundaries_parent,
+                                 mortars_parent,
+                                 mesh,
+                                 equations,
+                                 dg,
+                                 ::Type{uEltype}) where {uEltype <: Real}
+    elements = deepcopy(elements_parent)
+    resize!(elements, length(mesh.cell_ids))
+
+    @views elements.inverse_jacobian .= elements_parent.inverse_jacobian[..,
+                                                                         mesh.cell_ids]
+    @views elements.jacobian_matrix .= elements_parent.jacobian_matrix[..,
+                                                                       mesh.cell_ids]
+    @views elements.node_coordinates .= elements_parent.node_coordinates[..,
+                                                                         mesh.cell_ids]
+    @views elements.contravariant_vectors .= elements_parent.contravariant_vectors[..,
+                                                                                   mesh.cell_ids]
+    @views elements.surface_flux_values .= elements_parent.surface_flux_values[..,
+                                                                               mesh.cell_ids]
+    interfaces = extract_interfaces(mesh, interfaces_parent)
+
+    return elements, interfaces, boundaries_parent, mortars_parent
+end
+
+function extract_interfaces(mesh::P4estMeshView, interfaces_parent)
+    # Remove all interfaces that have a tuple of neighbor_ids where at least one is
+    # not part of this meshview, i.e. mesh.cell_ids.
+
+    mask = BitArray(undef, ninterfaces(interfaces_parent))
+    for interface in 1:size(interfaces_parent.neighbor_ids)[2]
+        mask[interface] = (interfaces_parent.neighbor_ids[1, interface] in mesh.cell_ids) &&
+                          (interfaces_parent.neighbor_ids[2, interface] in mesh.cell_ids)
+    end
+    interfaces = deepcopy(interfaces_parent)
+    interfaces = P4estInterfaceContainer(interfaces_parent.u,
+                                         interfaces_parent.neighbor_ids,
+                                         interfaces_parent.node_indices,
+                                         interfaces_parent._u,
+                                         interfaces_parent._neighbor_ids,
+                                         interfaces_parent._node_indices)
+    resize!(interfaces, sum(mask))
+    interfaces.u = interfaces_parent.u[.., mask]
```

Comment:
```suggestion
    @views interfaces.u .= interfaces_parent.u[.., mask]
```
Required here and below, as you did it for the element container above